Sming/Services/SpifFS/spiffs_sming.c

```diff
@@ -110,35 +118,64 @@ static void spiffs_mount_internal(spiffs_config *cfg)
   cfg->hal_read_f = api_spiffs_read;
   cfg->hal_write_f = api_spiffs_write;
   cfg->hal_erase_f = api_spiffs_erase;
-  
-  uint32_t dat;
-  bool writeFirst = false;
-  flashmem_read(&dat, cfg->phys_addr, 4);
-  //debugf("%X", dat);
 
-  if (dat == UINT32_MAX)
-  {
-	  debugf("First init file system");
-	  spiffs_format_internal(cfg);
-	  writeFirst = true;
+  int res = spiffs_mount_minimal(cfg);
+  debug_d("Mount result: %d\n", res);
+
+  if(res < 0) {
+	  int check = SPIFFS_check(&_filesystemStorageHandle);
+	  if(check < 0) {
+		  debug_w("Unsuccessful SPIFFS check with error %d.\r\n", check);
+		  debug_w("ALL DATA WILL BE ERASED. First init filesystem.\r\n");
+
+#if(SPIFFS_USE_MAGIC == 1)
+		  // See https://github.com/pellepl/spiffs/wiki/Using-spiffs - Formatting
+		  debug_d("SPIFFS_USE_MAGIC is set!\r\n");
+		  if(res == SPIFFS_ERR_NOT_A_FS) {
+			  debug_d("Error SPIFFS_ERR_NOT_A_FS which is expected, continuing with format...\r\n");
+			  spiffs_unmount();
+			  spiffs_format_internal(cfg);
+		  }
+#else
+		  debug_i("Formatting fs...\r\n");
+		  spiffs_unmount();
+		  spiffs_format_internal(cfg);
+#endif
+		  res = spiffs_mount_minimal(cfg);
+		  debug_i("Mount result is: %d\r\n", res);
```

Comment:
Remove newlines from _debug_*_ strings. Also in the other _debug__ lines introduced in this PR.